### PR TITLE
training bug fix and adding launch script for speech_multitask

### PR DESCRIPTION
**Pull request by krishnacpuvvada**
# What does this PR do ?
Fixes some bugs related to training after canary-2 refactor
Adds a launch script for AED multitask models in `examples/asr/speech_multitask` folder

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: examples/asr/speech_multitask/speech_to_text_aed.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    model.train_ds.audio.tarred_audio_filepaths=<path to tar files with audio> \
+    model.train_ds.audio_manifest_filepath=<path to audio data manifest> \
+    model.validation_ds.manifest_filepath=<path to validation manifest> \
+    model.test_ds.manifest_filepath=<path to test manifest> \
+    model.tokenizer.dir=<path to directory of tokenizer (not full path to the vocab file!)> \
+    model.tokenizer.model_path=<path to speech tokenizer model> \
+    model.tokenizer.type=<either bpe, wpe, or yttm> \
+    trainer.devices=-1 \
+    trainer.accelerator="ddp" \
+    trainer.max_epochs=100 \
```
Comment (pzelasko) [inline examples/asr/speech_multitask/speech_to_text_aed.py:29]:
```suggestion
    trainer.max_steps=100000 \
```

Comment (pzelasko) [reply examples/asr/speech_multitask/speech_to_text_aed.py:29]: I'd also add the following options here (or consider adding them to the config):

```
+trainer.limit_train_batches=20000
trainer.val_check_interval=5000
+trainer.use_distributed_sampler=false
```

File: nemo/collections/asr/models/aed_multitask_models.py [hunk 2]
```diff
@@ -81,6 +81,11 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         # Setup the tokenizer
         self._setup_tokenizer(cfg.tokenizer)
 
+        # Assert config has "prompt_format"
+        if "prompt_format" not in cfg:
```
Comment (pzelasko) [inline nemo/collections/asr/models/aed_multitask_models.py:85]: should we add it to config with `???` and in the documentation of `speech_to_text_aed.py` script?

File: examples/asr/speech_multitask/speech_to_text_aed.py (hunk 1 above)
Comment (titu1994) [reply examples/asr/speech_multitask/speech_to_text_aed.py:29]: Why train limit batches?

File: nemo/collections/asr/models/aed_multitask_models.py (hunk 2 above)
Comment (titu1994) [reply nemo/collections/asr/models/aed_multitask_models.py:85]: I did add it to config with Canary as default I think

**Review by titu1994 — APPROVED** [review]
Very nice changes, lgtm

**Review by stevehuang52 — APPROVED** [review]
LGTM, thanks~

File: examples/asr/speech_multitask/speech_to_text_aed.py (hunk 1 above)
Comment (krishnacpuvvada) [reply examples/asr/speech_multitask/speech_to_text_aed.py:29]: comment from @pzelasko: without limit_train_batches code flow triggers `len(dataset)`.  when using lhotse, due to dynamic batch sizes, you're sampling for a total amount of duration, the `len(dataset)` is different for every random seed and not known up front.

File: nemo/collections/asr/models/aed_multitask_models.py (hunk 2 above)
Comment (krishnacpuvvada) [reply nemo/collections/asr/models/aed_multitask_models.py:85]: yes, canary is default in yaml config.

**Review by pzelasko — APPROVED** [review]

**Pull request merged**